presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ActualProperties.java

```diff
@@ -167,6 +187,13 @@ public ActualProperties translate(Function<Symbol, Optional<Symbol>> translator)
         return localProperties;
     }
 
+    public ActualProperties withReplicatedNulls(boolean replicatedNulls)
```

Comment:
its also used to propagate existing replicated nulls too (which can be true or false). There might be a better way to make that naturally happen however.